lib/components/CircuitJsonPreview/CircuitJsonPreview.tsx

```diff
@@ -133,6 +133,35 @@ export const CircuitJsonPreview = ({
     }
   }, [circuitJson])
 
+  // Detect Dark Reader dark mode using data-darkreader-scheme
+  const [isDarkReaderDark, setIsDarkReaderDark] = useState(
+    () =>
+      document.documentElement.getAttribute("data-darkreader-scheme") ===
+      "dark",
+  )
+
+  useEffect(() => {
+    let frameId: number
+    let lastValue = isDarkReaderDark
+
+    const check = () => {
+      const current =
+        document.documentElement.getAttribute("data-darkreader-scheme") ===
+        "dark"
+      if (current !== lastValue) {
+        setIsDarkReaderDark(current)
+        lastValue = current
+      }
+      frameId = requestAnimationFrame(check)
+    }
+
+    frameId = requestAnimationFrame(check)
+
+    return () => {
+      cancelAnimationFrame(frameId)
+    }
+  }, [])
```

Comment:
hehe this runs really frequently and at the very least should be inside it's own hook (i keep telling people to move things to avoid giant component files, it's making me think i should just have an LoC limit on each file or something, i get that you might just be showing the approach) `useIsInversionDarkMode` is a good hook name- `useIsDarkMode` doesn't make sense because we're detecting if something is inverting the page.

Also I'm trying to avoid anything dark-reader specific, we want to be able to detect dark mode more generally, e.g. chrome has a builtin setting that seems to do the same transformations as dark reader.